tests/dummy/app/templates/dialog/dialog-modal.hbs

```diff
@@ -0,0 +1,63 @@
+<div class="flex items-start justify-center w-screen h-full p-12 bg-gray-50">
+  {{#let (hash isOpen=false) as |state|}}
+    <button
+      type="button"
+      class="inline-flex items-center px-4 py-2 text-base font-medium text-white bg-indigo-600 border border-transparent rounded-md shadow-sm hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
+      {{on "click" (set state.isOpen true)}}
+    >
+      Open dialog
+    </button>
+
+    <Dialog
+      class="fixed inset-0 z-10 overflow-y-auto"
+      @as="div"
+      @open={{state.isOpen}}
+      @onClose={{set state.isOpen false}}
+      as |d|
+    >
+      <div class="flex items-end justify-center min-h-screen px-4 pt-4 pb-20 text-center sm:block sm:p-0">
+        <d.Overlay class="fixed inset-0 bg-gray-500 bg-opacity-75 transition-opacity" />
+
+        {{!-- This element is to trick the browser into centering the modal contents. --}}
+        <span class="hidden sm:inline-block sm:align-middle sm:h-screen" aria-hidden="true">
+          &#8203;
+        </span>
+
+        <div class="inline-block px-4 pt-5 pb-4 overflow-hidden text-left align-bottom bg-white rounded-lg shadow-xl transform transition-all sm:my-8 sm:align-middle sm:max-w-sm sm:w-full sm:p-6">
+          <div>
+            <div class="flex items-center justify-center w-12 h-12 mx-auto bg-green-100 rounded-full">
+              <svg xmlns="http://www.w3.org/2000/svg" class="w-6 h-6 text-green-600" fill="none" viewBox="0 0 24 24" stroke="currentColor">
+                <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M5 13l4 4L19 7" />
+              </svg>
+            </div>
+
+            <div class="mt-3 text-center sm:mt-5">
+              <d.Title
+                class="text-lg font-medium text-gray-900 leading-6"
+                @as="h3"
+              >
+                Payment successful
+              </d.Title>
+
+              <div class="mt-2">
+                <p class="text-sm text-gray-500">
+                  Lorem ipsum dolor sit amet consectetur adipisicing elit. Consequatur amet labore.
+                </p>
+              </div>
+            </div>
+          </div>
+
+          <div class="mt-5 sm:mt-6">
+            <button
+              type="button"
+              class="inline-flex justify-center w-full px-4 py-2 text-base font-medium text-white bg-indigo-600 border border-transparent rounded-md shadow-sm hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500 sm:text-sm"
+              {{on "click" (set state.isOpen false)}}
```

Comment:
perhaps this would be slightly better as `{{on "click" dialog.onClose}}`?

Reply:
The `<Dialog/>` component doesn't yield the `onClose` action. Would you like it too?

I'm kind of figuring that what you pointed out here is sort of down to the simplistic implementation of the dummy app - I was just trying to keep the logic in the template. A real world implementation might have a `closeModal` action on the component or whatever, that could feasibly used when passed in to the Dialog and on any buttons on the modal.

I'm not overly precious about this - it's just the dummy app.

If you think the `<Dialog />` component should expose the `@onClose` action though, I can do that. Just can't, yet, see the benefit myself. But I'm probably missing something :)

Reply:
Done 👍

Reply:
> I was just trying to keep the logic in the template

I think that's a good reason to yield the `onClose` action, it allows for the action to be defined in one place. My default is to use template only components, I'm sure that I'd use this myself.

Reply:
Okie doke. I'll make the change.

Reply:
Done 👍